File: AcceuilWayfinding/app/src/main/java/e/plass/acceuilwayfinding/model/Network/RetrofitInstance.java
package e.plass.acceuilwayfinding.model.Network;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitInstance {
    private static       Retrofit retrofit;
    private static final String   BASE_URL = "https://vitrine123.000webhostapp.com/Service_Way/Controller/";

    public static Retrofit getRetrofitInstance() {
        if (retrofit == null) {
            retrofit = new retrofit2.Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }
}

File: app/src/main/java/com/example/wayfindingdev/Network/RetrofitInstance.kt
package com.example.wayfindingdev.Network

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitInstance {
    private var retrofit: Retrofit? = null
    private val BASE_URL = "https://wayfinding.chickdev.space/api/Controller/"

    val retrofitInstance: Retrofit
        get() {
            if (retrofit == null) {
                retrofit = Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
            }
            return retrofit!!
        }
}
